parsing/lexer.mll

```diff
@@ -300,6 +301,13 @@ let prepare_error loc = function
       Location.errorf ~loc
         "This comment contains an unterminated string literal"
         ~sub:[Location.msg ~loc:literal_loc "String literal begins here"]
+  | Empty_character_literal ->
+      Location.errorf ~loc
+        "%a"
+        Format.pp_print_text
+        "The empty character literal '' is invalid. \
+         We expect a character between single quotes (possibly a space ' ') \
+         or a single quote for type variables 'a."
```

Comment:
Over here, using "we" when you are singular is generally reserved for the monarch 🙂 However, I think it's more consistent with the other messages to describe the properties of what is expected:
```suggestion
        "Character expected between two single quotes (' ', for example). \
         The empty character literal '' is invalid. \
         A single quote is used for type variables 'a."
```

Reply:
I'm unconvinced by your proposition: I have doubts that "Character expected ... ." is easier to read than "We expect a character ... .", and your message splits the allowed forms in two halves, one before and one after the diagnostic, I find this confusing.

If the `We` is really a problem (we, the authors of the compiler), I can change to "Invalid empty character literal ''.", but I would rather keep this structure -- or something else. But then there are many occurrences of "we" in the compiler text, mostly in comments or in the manual, but also a handful in error messages. (`git grep ' [Ww]e ' | wc -l` returns 2440 on my copy of trunk.)

Reply:
> If the `We` is really a problem (we, the authors of the compiler),

Personally except for the notable exception of *did you mean*-like error messages, I dislike the anthropomorphisation of computer program error messages.  

Also there are already errors patterns in the character space with which consistency could be mainatined e.g. this use of "illegal" vs "invalid".
```
# "bl\a";;
Warning 14: illegal backslash escape in string.
- : string = "bl\\a"
```
Here would be my proposal:
```
illegal empty character literal ''. Did you mean ' ' ? 
```

Reply:
The use of "we" in the manual or comments is entirely distinct from messages displayed by the tools. We may write the compiler, but on encountering an error the message comes from the compiler, not its authors. There may be other error messages elsewhere in the compiler which do this, but the other lexer error messages right around this one do not.

@dbuenzli's suggestion looks much better to me.

Reply:
Okay, I went with

```
Error: Invalid empty character literal; did you mean ' ' or a type variable 'a ?
```

Reply:
I would make the hint part more distinct:
```ocaml
Error: Illegal empty character literal
Hint: Did you mean ' ' or a type variable 'a ?
```
(With the hope of one day using pretty printer for hints).

Reply:
> Error: Illegal empty character literal; did you mean ' ' or a type variable 'a ?

Sorry for the extreme hci bikesheding @gasche. 

Since the original report is about the difficulty to distinguish spaces with certain fonts I think it would be good to write down the empty `''` as was  in your your original message and in my proposal. 

That way a direct, local, comparison between `''` and `' '` can be made which should easily show the difference; the context switch to the actual code may make you loose the distinction.

Also I notice the existing format for reporting `Did you mean` is slightly different:
```ocaml
# let fab x = x;;
val fab : 'a -> 'a = <fun>
# fib;;
Error: Unbound value fib
Hint: Did you mean fab?
```
Not that I'm necessarily in favor of that one. However I'm slightly averse to have the *did you mean* after a semicolon, rather than a `.`.

Reply:
In this case I don't mind the bikeshedding, especially given that you all had good reasons to suggest changes. I will follow the standard "did you mean" format suggested by @Octachron, but include the `''` back. (I agree it's interesting to show it.)